training/checkpoint.py

```diff
@@ -36,7 +46,42 @@ def load_checkpoint(checkpoint_path, model, optimizer, train_loader):
     return model, optimizer, iteration, epoch
 
 
-def warm_start_model(checkpoint_path, model, ignore_layers=["embedding.weight"]):
+def transfer_symbols_embedding(original_embedding_weight: torch.Tensor, embedding_layer, original_symbols: Optional[list], new_symbols: list):
+    if original_symbols is None:
+        original_symbols = NVIDIA_ALPHABET
+    assert len(original_symbols) == original_embedding_weight.shape[0], f'length of original_symbols does not match length of checkpoint model embedding! Got {len(original_symbols)} and {original_embedding_weight.shape[0]}.'
+    
+    weight_tensor = original_embedding_weight.data
+    original_std = weight_tensor.std()
+    original_mean = weight_tensor.mean()
+    
+    weight_dict = {}
+    for symbol_index, symbol in enumerate(original_symbols):
+        weight_dict[symbol] = weight_tensor[symbol_index] # save vector for each symbol into it's own key
+    
+    for symbol_index, new_symbol in enumerate(new_symbols):
+        # transfers matching symbols from pretrained model to new model  e.g: 'e' -> 'e'
+        if new_symbol in weight_dict:
+            embedding_layer.weight.data[symbol_index] = weight_dict[new_symbol]
+        
+        # transfers non-ascii symbols from pretrained model to new model  e.g: 'e' -> 'é'
+        elif unidecode(new_symbol) in weight_dict:
+            embedding_layer.weight.data[symbol_index] = weight_dict[unidecode(new_symbol)]
+        
+        # transfers upper-case symbols from pretrained model to new model  e.g: 'E' -> 'e'
+        elif new_symbol.upper() in weight_dict:
+            embedding_layer.weight.data[symbol_index] = weight_dict[new_symbol.upper()]
+        
+        # transfers lower-case symbols from pretrained model to new model  e.g: 'e' -> 'E'
```

Comment:
Do we need to convert to uppercase?

Reply:
I included all conversion cases I could think of just to make sure it always works.